language/movie_nl/main.py

```diff
@@ -0,0 +1,366 @@
+    """Analyze the document and get the distribution of sentiments and
+    the movie name."""
+    logging.info('Analyzing {}'.format(document.doc_id))
+
+    sentences, entities = document.extract_all_sentences(service)
+
+    try:
+        sentiments = [get_sentiment(service, sentence) for sentence in sentences]
+    except HttpError as e:
+        raise e
+
+    return sentiments, entities
+
+
+def get_request_body(text, syntax=True, entities=True, sentiment=True):
+    """Creates the body of the request to the language api in
+    order to get an appropriate api response."""
+    body = {
+        'document': {
+            'type': 'PLAIN_TEXT',
+            'content': text,
+        },
+        'features': {
+            'extract_syntax': syntax,
+            'extract_entities': entities,
+            'extract_document_sentiment': sentiment,
+        },
+        'encoding_type': 'UTF32'
+    }
+
+    return body
+
+def get_response_with_retry(request, tries, retry=3):
+    """Get the response using re-try"""
+    try:
+        response = request.execute()
+        return response
+    except HttpError as e:
+        raise e
+    except socket.error as se:
+        if tries > retry:
+            raise se
+
+        logging.error('Re-trying the request {}'.format(tries))
+
+        tries+=1
+        return get_response_with_retry(request, tries, retry)
+
+
+def get_sentiment(service, sentence):
+    """Get the sentence-level sentiment."""
+    body = get_request_body(
+        sentence, syntax=False, entities=True, sentiment=True)
+
+    docs = service.documents()
+    request = docs.annotateText(body=body)
+
+    response = get_response_with_retry(request, 1)
+
+    sentiment = response.get("documentSentiment")
+
+    if sentiment is None:
+        return (None, None)
+    else:
+        pol = sentiment.get("polarity")
+        mag = sentiment.get("magnitude")
+
+    if pol is None and mag is not None:
+        pol = 0
+    return (pol, mag)
+
+
+class Document(object):
+    """Document class captures a single document of movie reviews."""
+
+    def __init__(self, text, doc_id, doc_path):
+        self.text = text
+        self.doc_id = doc_id
+        self.doc_path = doc_path
+        self.sentent_pair = None
+        self.label = None
+
+    def extract_all_sentences(self, service):
+        """Extract the sentences in a document."""
+
+        if self.sentent_pair is None:
+            docs = service.documents()
+            request_body = get_request_body(
+                self.text,
+                syntax=True,
+                entities=True,
+                sentiment=False)
+            request = docs.annotateText(body=request_body)
+
+            ent_list = []
+
+            response = request.execute()
+            entities = response.get('entities', [])
+            sentences = response.get('sentences', [])
+
+            sent_list = [
+                sentence.get('text').get('content') for sentence in sentences
```

Comment:
sentence.get('text', {}).get...

If you're using `.get`, it means you might be getting `None`, which will error if you try to `.get` it.